### ISSUE TITLE
Add Nextcloud sync to Android app

### ISSUE THREAD

**Issue by blacklightpy** [CollaboraOnline/online#7134]
### Is your feature request related to a problem?

The Android app doesn't seem to have Nextcloud sync. It works on the Nextcloud app, but users may expect the sync functionality to be in the official app.

### Describe the solution you'd like

Collabora Android app should have a sync functionality. It can either use Collabora's cloud if they have paid a subscription, or they can enter an alternate server.

Microsoft has Onedrive and Office 365. The files cannot be edited in the Onedrive app, and requires an external app like Office 365 to edit it. Similarly, Nextcloud can have an open with an external app feature (which it doesn't have now for documents), with a recommendation for installing Collabora. This is not an issue on the web client, as the redirection can happen automatically. This shouldn't require much aside from implementing a Sign in to Nextcloud section in the main app. Of course, nothing prevents me from opening the app from Nextcloud via the file picker, but since Google and Microsoft have offered a seamless integration, the user expectation would be to have a sign-in button.

### Describe alternatives you've considered

Or, Nextcloud has a good editor, but one isn't likely to find that it exists without hearing it from elsewhere. So, maybe have a wrapper app like MS Office 365, which just lets you make Nextcloud documents and use the editor from the Nextcloud app. That would be as if Office 365 requires standalone editor apps to be downloaded to get the editing functionality, which was how it used to be IIRC. But this would be bad from a user expectation standpoint, I guess.

### Additional context

N/A

**Comment by sebastiaanveld** [comment]
My take on this is free vs enterprise: to get into the integrations area one probably needs or wants support on that too so it keeps working correctly. So, get an CO App Enterprise subscription or like a payed version of the app from the store. That would mean two apps, a free one and a paid one which also would IMO need to have mobile app management integration so admins can manage what users can to with the app (which integrations to enable or not, where files can be saved or imported from, how to authenticate, corporate branding options, default settings etc). Think that this is a long term goal maybe (@mmeeks ?). I still envision (prefer) a "free" version be a LibreOffice branded one to get more exposure and the CO App be the enterprise one with those kind of enterprise integration features in it. That would make it possible for like Nextcloud to leverage the CO App (as part of their subscription and working with Collabora) and do the needed integrations and defaults their way.

**Comment by mmeeks** [comment]
Interesting ticket. I read this as a request to bundle Collabora Office for Android (COOLA) into the relevant Nextcloud app. Of course the Nextcloud app has built in support for using Collabora Online (the same UI but collaboratively editing inside a nested Webview) for opening files directly in the Nextcloud app: so we're good there already. The Nextcloud app can also use a native COOLA on device to open files off-line (using normal OS editing mechanisms). COOLA can also browse Nextcloud folders using the built in OS folder browsing - which is not very lovely to be sure but ... at least something.

The idea of bundling Nextcloud & COOLA and calling it 'LibreOffice' would be a travesty from a crediting, as well as a business/lead flow perspective - sorry @sebastiaanveld =) If anything we could call it either CODE or Nextcloud - but all our mobile versions are zero price anyway: and if you want you can pay for support separately for the same binary.

Ultimately I think this is in large part a marketing and product positioning problem; Collabora 'Office' - doesn't look like Microsoft Office - which is in reality a combination of a cloud storage like Nextcloud and a (much more cut down) Office suite.

So - I don't see us solving this one really - unless some volunteer wants to graunch the two together - but of course COOLA also works nicely with ownCloud, and other EFSS'. If there is some better way of presenting a UI that works nicely without duplicating the local-synching and the UI generally there that would be really lovely. Its not trivial writing a mobile app to do what any of those synching apps do - even if the UI looks superficially simple.

Thanks !

**Comment by blacklightpy** [comment]
There is one problem that needs to be fixed. Collabora documents can't be edited using the Collabora Office Android app.

In the Nextcloud app, files like PDF have an "open with" context menu option. For recognized files like PNG, it opens directly and has no such option, for MP4 it has a stream with option. And for LibreOffice documents too it does not have an open with option.

I suppose this is part of the Nextcloud Collabora integration. Not sure if this should be reported here or in Nextcloud for Android.

Also, this would solve the part of my problem mentioned here:
> Microsoft has Onedrive and Office 365. The files cannot be edited in the Onedrive app, and requires an external app like Office 365 to edit it. Similarly, Nextcloud can have an open with an external app feature (which it doesn't have now for documents), with a recommendation for installing Collabora. This is not an issue on the web client, as the redirection can happen automatically. This shouldn't require much aside from implementing a Sign in to Nextcloud section in the main app.

> The Nextcloud app can also use a native COOLA on device to open files off-line (using normal OS editing mechanisms). 

Nope. That's the problem.

Also I made a typo here, that's probably what made you overlook that.
Of course, nothing prevents me from opening the ~~app~~ **documents** from Nextcloud via the file picker, but since Google and Microsoft have offered a seamless integration, the user expectation would be to have a sign-in button.

As for bundling Nextcloud and Collabora for Android into one app:

> The idea of bundling Nextcloud & COOLA and calling it 'LibreOffice' would be a travesty from a crediting, as well as a business/lead flow perspective - sorry @sebastiaanveld =) If anything we could call it either CODE or Nextcloud

1. I don't think that's what he meant. I took it as he said the current version of Collabora Office for Android should be changed to LibreOffice, since it doesn't offer any other features. And a version with the sync to Collabora should be branded as Collabora Office, based on LibreOffice. I think this looks amazing from a branding perspective.

2. And that was exactly my idea too - to add Nextcloud sync (just for documents) into Collabora Office as an upgrade. That would make more sense for the app being Collabora Office rather than LibreOffice.
